music.js

```diff
@@ -1,17 +1,28 @@
-// Define the Artist constructor function here, with a 'name' property that can be set at instantiation,
-// and a 'songs' property that is an empty array to start
-function Artist() {
-
+function Artist(name) {
+  this.name = name;
+  this.songs = [];
 }
 
-// Define the Song constructor function here, with 'title' and 'artist' properties that can be set at instantiation,
-// and a 'playCount' property that is an empty array to start
-function Song() {
-
-  // When a new song is instantiated, you'll want to add that song to its artist's 'songs' array.
-  // You'll probably want to do that in here somewhere...
+function Song(title, artist) {
+  this.title = title;
+  this.artist = artist;
+  this.playCount = 0;
+  Artist.prototype.add = function(title) {
+    this.songs.push(title)
+  }
+  artist.add(title);
 }
 
+// function play(title) {
+//   Song.prototype.add = function() {
```

Comment:
So aside from this code being commented out, this is probably not what you wanna do. First of all, you don't want this `Song.prototype.add` method to be defined within another function - all that is going to do is create another step before you even are able to *define* the method `add` on the `Song` constructor.

Third, we don't want the `play` method to exist just out in the open here - we want it to be a method of the `Song` object. And the `title` argument here isn't really helping us find a specific song we want to update.

Instead this whole block of code should probably look like this instead:

```
Song.prototype.play = function() {
  this.playCount++;
}

var vanHalen = new Artist("Van Halen");
var jump = new Song("Jump!", vanHalen);

jump.play();
```